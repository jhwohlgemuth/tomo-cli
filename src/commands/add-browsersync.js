import {
    PackageJsonEditor,
    allDoExist,
    install,
    someDoExistSync,
    uninstall
} from '../api';

const BROWSERSYNC_DEPENDENCIES = [
    'browser-sync',
    'npm-run-all'
];
/**
 * @type {task[]}
 * @see https://www.browsersync.io/docs/command-line
 */
export const addBrowsersync = [
    {
        text: 'Add Browsersync tasks to package.json',
        task: async ({outputDirectory, port}) => {
            const scripts = {
                prestart: 'npm run build:es',
                start: `npm-run-all --parallel watch:es watch:css serve`,
                serve: `browser-sync start --server ${outputDirectory} --files ${outputDirectory} --port ${port}`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: ({useParcel}) => allDoExist('package.json', 'postcss.config.js') && (someDoExistSync('webpack.config.js', 'rollup.config.js') || useParcel) // eslint-disable-line max-len
    },
    {
        text: 'Install Browsersync dependencies',
        task: ({skipInstall}) => install(BROWSERSYNC_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline, skipInstall}) => !skipInstall && isNotOffline && allDoExist('package.json')
    }
];
export const removeBrowsersync = [
    {
        text: 'Remove Browsersync tasks from package.json',
        task: async () => {
            const scripts = {
                prestart: undefined,
                start: undefined,
                serve: undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall Browsersync dependencies',
        task: () => uninstall(BROWSERSYNC_DEPENDENCIES),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...BROWSERSYNC_DEPENDENCIES),
        optional: ({skipInstall}) => !skipInstall
    }
];
export default addBrowsersync;