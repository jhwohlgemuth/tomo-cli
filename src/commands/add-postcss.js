import {
    PackageJsonEditor,
    PostcssConfigEditor,
    allDoExist,
    allDoNotExist,
    install,
    uninstall
} from '../api';

const POSTCSS_DEPENDENCIES = [
    'cssnano',
    'postcss',
    'postcss-cli',
    'postcss-reporter',
    'postcss-safe-parser',
    'postcss-import',
    'postcss-preset-env',
    'stylelint',
    'stylelint-config-recommended'
];
/**
 * @type {task[]}
 * @see https://github.com/postcss/postcss
 */
export const addPostcss = [
    {
        text: 'Create PostCSS config file',
        task: async ({useSnowpack}) => {
            const plugins = [
                `require('stylelint')({config: {extends: 'stylelint-config-recommended'}})`,
                `require('postcss-import')()`,
                `require('postcss-preset-env')({stage: 0})`,
                `require('cssnano')()`,
                `require('postcss-reporter')({clearReportedMessages: true})`
            ];
            await (new PostcssConfigEditor())
                .create()
                .extend({plugins})
                .extend(useSnowpack ? {map: 'false'} : {})
                .commit();
        },
        condition: () => allDoNotExist('postcss.config.js')
    },
    {
        text: 'Add PostCSS tasks to package.json',
        task: async ({assetsDirectory, outputDirectory}) => {
            const scripts = {
                'build:css': `postcss ${assetsDirectory}/css/style.css --dir ${outputDirectory}`,
                'watch:css': 'npm run build:css -- --watch'
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Install PostCSS dependencies',
        task: ({skipInstall}) => install(POSTCSS_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json')
    }
];
export const removePostcss = [
    {
        text: 'Delete PostCSS config file',
        task: async () => {
            await (new PostcssConfigEditor())
                .delete()
                .commit();
        },
        condition: () => allDoExist('postcss.config.js')
    },
    {
        text: 'Remove PostCSS build task from package.json',
        task: async () => {
            const scripts = {
                'build:css': undefined,
                'watch:css': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall PostCSS dependencies',
        task: () => uninstall(POSTCSS_DEPENDENCIES),
        condition: () => allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...POSTCSS_DEPENDENCIES)
    }
];
export default addPostcss;