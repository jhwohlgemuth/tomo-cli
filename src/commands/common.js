import {join} from 'path';
import {mkdirp} from 'fs-extra';
import {
    PackageJsonEditor,
    Scaffolder,
    allDoNotExist
} from '../api';

/** @ignore */
export const createEditorConfiguration = [
    {
        text: 'Create editor configuration file',
        task: async ({overwrite}) => {
            await (new Scaffolder(join(__dirname, 'common', 'templates')))
                .overwrite(overwrite)
                .target('.')
                .copy('.editorconfig', '.editorconfig')
                .commit();
        },
        condition: () => allDoNotExist('.editorconfig')
    }
];
/** @ignore */
export const createPackageJson = [
    {
        text: 'Create package.json',
        task: async () => {
            await (new PackageJsonEditor())
                .create()
                .commit();
        },
        condition: () => allDoNotExist('package.json')
    }
];
/** @ignore */
export const createSourceDirectory = [
    {
        text: 'Create source directory',
        task: ({sourceDirectory}) => mkdirp(sourceDirectory),
        condition: ({sourceDirectory}) => allDoNotExist(sourceDirectory)
    }
];