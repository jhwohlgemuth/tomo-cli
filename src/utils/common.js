import {join} from 'path';
import {is} from 'ramda';
import {pathExists, pathExistsSync} from 'fs-extra';
import prettier from 'prettier';

const PRETTIER_OPTIONS = {
    bracketSpacing: false,
    parser: 'json-stringify',
    printWidth: 150,
    tabWidth: 4,
    singleQuote: true
};
const newMap = val => new Map(val);
const joinPath = name => join(process.cwd(), name);
const checkPathExists = name => name |> joinPath |> pathExists;
const checkPathExistsSync = name => name |> joinPath |> pathExistsSync;
export const dict = val => val |> Object.entries |> newMap;
export const isEmptyString = data => typeof data === 'string' && data.length === 0;
export const parse = data => data |> JSON.stringify |> JSON.parse;
export const maybeApply = (val, options) => is(Function)(val) ? val(options) : val;
export const getBinDirectory = path => {
    const [packageDirectory] = path.split('Makefile');
    return `${packageDirectory}node_modules/.bin/`;
};
/**
 * Format input code using Prettier
 * @param {*} [code=''] Code to be formatted
 * @example <caption>Prettier options</caption>
 * {
 *     bracketSpacing: false,
 *     parser: 'json-stringify',
 *     printWidth: 150,
 *     tabWidth: 4,
 *     singleQuote: true
 * }
 * @return {string} Code formatted by Prettier
 */
export const format = (code = {}) => prettier
    .format(JSON.stringify(code), PRETTIER_OPTIONS)
    .replace(/"/g, '');
/**
 * Check that at least one file or files exist
 * @param  {...string} args File or folder path(s)
 * @example
 * // some/folder/
 * //   ├─ foo.js
 * //   └── bar.js
 * const hasFoo = someDoExist('some/folder/foo.js');
 * const hasBaz = someDoExist('some/folder/baz.js');
 * const hasSomething = someDoExist('some/folder/bar.js', 'some/folder/baz.js');
 * console.log(hasFoo); // true
 * console.log(hasBaz); // false
 * console.log(hasSomething); // true
 * @return {boolean} Some files/path do exist (true) or all do not exist (false)
 */
export const someDoExist = async (...args) => {
    const checks = await Promise.all(args.map(checkPathExists));
    return checks.some(Boolean);
};
/**
 * Check that at least one file or files exist (synchronous version of {@link someDoExist})
 * @param  {...string} args File or folder path(s)
 * @return {boolean} Some files/path do exist (true) or all do not exist (false)
 */
export const someDoExistSync = (...args) => args.map(checkPathExistsSync).some(Boolean);
/**
 * Check that all files exist
 * @param  {...string} args File of folder paths
 * @return {boolean} All files/paths exist (true) or do not (false)
 */
export const allDoExist = async (...args) => {
    const checks = await Promise.all(args.map(checkPathExists));
    return checks.every(Boolean);
};
/**
 * Check that all files exist (synchronous version of {@link allDoExist})
 * @param  {...string} args File of folder paths
 * @return {boolean} All files/paths exist (true) or do not (false)
 */
export const allDoExistSync = (...args) => args.map(checkPathExistsSync).every(Boolean);
/**
 * Check that all files do not exist
 * @example
 * // some/folder/
 * //   ├─ foo.js
 * //   └── bar.js
 * const noPackageJson = allDoNotExist('some/folder/package.json');
 * console.log(noPackageJson); // true
 * @param  {...string} args File or folder path(s)
 * @return {boolean} All files/paths do not exist (true) or some do (false)
 */
export const allDoNotExist = async (...args) => {
    const checks = await Promise.all(args.map(checkPathExists));
    return checks.every(val => !val);
};
/**
 * Check that all files do not exist (synchronous version of {@link allDoNotExist})
 * @param  {...string} args File or folder path(s)
 * @return {boolean} All files/paths do not exist (true) or some do (false)
 */
export const allDoNotExistSync = (...args) => args.map(checkPathExistsSync).every(val => !val);