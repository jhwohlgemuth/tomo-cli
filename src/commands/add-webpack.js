import {join} from 'path';
import {oneLineTrim} from 'common-tags';
import {PackageJsonEditor, WebpackConfigEditor, install, uninstall} from '../utils';
import {allDoExist, allDoExistSync, allDoNotExist} from '../utils/common';

const DEPLOY_SCRIPTS = {
    predeploy: 'npm-run-all clean build:es build:css',
    deploy: 'echo \"Not yet implemented - now.sh or surge.sh are supported out of the box\" && exit 1'
};
const BUILD_DEPENDENCIES = [
    'cpy-cli',
    'del-cli',
    'npm-run-all'
];
const WEBPACK_DEPENDENCIES = [
    'webpack',
    'webpack-cli',
    'webpack-dashboard',
    'webpack-jarvis',
    'webpack-dev-server',
    'babel-loader',
    'terser-webpack-plugin'
];
const WEBPACK_RULES = [
    {
        test: `/\.jsx?$/`,
        exclude: `/node_modules/`,
        loader: `'babel-loader'`,
        query: {
            presets: [`'@babel/env'`]
        }
    }
];
const WEBPACK_RULES_WITH_CESIUM = [
    ...WEBPACK_RULES,
    {
        test: `/\.css$/`,
        use: [`'style-loader'`, `'css-loader'`]
    },
    {
        test: `/\.(png|gif|jpg|jpeg|svg|xml|json)$/`,
        use: [`'url-loader'`]
    }
];
const WEBPACK_PLUGINS = [
    `new DashboardPlugin()`
];
const WEBPACK_PLUGINS_WITH_CESIUM = [
    ...WEBPACK_PLUGINS,
    `new DefinePlugin({CESIUM_BASE_URL: JSON.stringify('/')})`,
    oneLineTrim`new CopyWebpackPlugin([
        {
            from: join(source, workers),
            to: 'Workers'
        },
        {
            from: join(source, 'Assets'),
            to: 'Assets'
        },
        {
            from: join(source, 'Widgets'),
            to: 'Widgets'
        }
    ])`
];
const CESIUM_ALIASES = {
    cesium$: `'cesium/Cesium'`,
    cesium: `'cesium/Source'`
};
const CESIUM_WEBPACK_DEPENDENCIES = [
    'copy-webpack-plugin',
    'css-loader',
    'style-loader',
    'url-loader'
];
const CESIUM_DEPENDENCIES = [
    ...CESIUM_WEBPACK_DEPENDENCIES,
    'cesium'
];
const RESIUM_DEPENDENCIES = [
    ...CESIUM_DEPENDENCIES,
    'resium'
];
/**
 * @type {task[]}
 * @see https://webpack.js.org/
 */
export const addWebpack = [
    {
        text: 'Create Webpack configuration file',
        task: async ({outputDirectory, port, sourceDirectory, withCesium, useReact}) => {
            const alias = {
                ...(useReact ? {'\'react-dom\'': `'@hot-loader/react-dom'`} : {}),
                ...(withCesium ? CESIUM_ALIASES : {})
            };
            const amd = {
                toUrlUndefined: true
            };
            const devServer = {
                port,
                contentBase: `'${outputDirectory}'`,
                compress: true,
                watchContentBase: true
            };
            const entryWithReact = [
                `'react-hot-loader/patch'`,
                `'${sourceDirectory}/main.js'`
            ];
            const entryWithoutReact = {
                app: `'${sourceDirectory}/main.js'`
            };
            const entry = useReact ? entryWithReact : entryWithoutReact;
            const optimization = {
                minimize: `argv.mode === 'production'`,
                minimizer: `[new TerserPlugin()]`
            };
            const node = {
                fs: `'empty'`
            };
            const plugins = withCesium ? WEBPACK_PLUGINS_WITH_CESIUM : WEBPACK_PLUGINS;
            const resolve = {
                modules: `[resolve(__dirname, '${sourceDirectory}'), 'node_modules']`,
                extensions: `[${useReact ? `'.js', '.jsx'` : `'.js'`}]`,
                alias
            };
            const rules = withCesium ? WEBPACK_RULES_WITH_CESIUM : WEBPACK_RULES;
            await (new WebpackConfigEditor())
                .create()
                .prepend(withCesium ? `const workers = '../Build/Cesium/Workers';` : '')
                .prepend(withCesium ? `const source = 'node_modules/cesium/Source';` : '')
                .prepend(`const TerserPlugin = require('terser-webpack-plugin');`)
                .prepend(`const DashboardPlugin = require('webpack-dashboard/plugin');`)
                .prepend(withCesium ? `const CopyWebpackPlugin = require('copy-webpack-plugin');` : '')
                .prepend(withCesium ? `const {DefinePlugin} = require('webpack');` : '')
                .prepend(`const {${withCesium ? 'join, ' : ''}resolve} = require('path');`)
                .prepend(`/* eslint-env node */`)
                .extend({devServer, entry, module: {rules}, optimization, plugins, resolve})
                .extend(withCesium ? {amd, node} : {})
                .commit();
        },
        condition: () => allDoNotExist('webpack.config.js')
    },
    {
        text: 'Add Webpack build tasks to package.json',
        task: async ({assetsDirectory, outputDirectory, sourceDirectory}) => {
            const scripts = {
                ...DEPLOY_SCRIPTS,
                clean: `del-cli ${outputDirectory}`,
                copy: 'npm-run-all --parallel copy:assets copy:index',
                'copy:assets': `cpy '${assetsDirectory}/!(css)/**/*.*' '${assetsDirectory}/**/[.]*' ${outputDirectory} --parents --recursive`,
                'copy:index': `cpy '${assetsDirectory}/index.html' ${outputDirectory}`,
                'prebuild:es': `del-cli ${join(outputDirectory, assetsDirectory)}`,
                'build:es': 'webpack',
                'postbuild:es': 'npm run copy',
                'watch:assets': `watch 'npm run copy' ${assetsDirectory}`,
                'watch:es': `watch 'npm run build:es' ${sourceDirectory}`,
                dashboard: 'webpack-dashboard -- webpack-dev-server --config ./webpack.config.js'
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Configure dev task',
        task: async ({skipInstall}) => {
            const scripts = {
                dev: 'stmux [ \"npm run dashboard\" : \"npm run lint:ing\" ]'
            };
            await install(['stmux'], {dev: true, skipInstall});
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json', '.eslintrc.js'),
        optional: () => allDoExistSync('package.json', '.eslintrc.js')
    },
    {
        text: 'Install Webpack and development dependencies',
        task: ({skipInstall}) => install([...BUILD_DEPENDENCIES, ...WEBPACK_DEPENDENCIES], {dev: true, skipInstall}),
        condition: ({isNotOffline, skipInstall}) => !skipInstall && isNotOffline && allDoExist('package.json')
    },
    {
        text: 'Install Cesium dependencies',
        task: ({skipInstall, useReact}) => install(useReact ? RESIUM_DEPENDENCIES : CESIUM_DEPENDENCIES, {skipInstall}),
        condition: ({withCesium}) => withCesium,
        optional: ({withCesium}) => withCesium
    }
];
export const removeWebpack = [
    {
        text: 'Delete Webpack configuration file',
        task: async () => {
            await (new WebpackConfigEditor())
                .delete()
                .commit();
        },
        condition: () => allDoExist('webpack.config.js')
    },
    {
        text: 'Remove Webpack build tasks from package.json',
        task: async () => {
            const scripts = {
                copy: undefined,
                'copy:assets': undefined,
                'copy:index': undefined,
                'watch:assets': undefined,
                dev: undefined,
                'prebuild:es': undefined,
                'build:es': undefined,
                'postbuild:es': undefined,
                'watch:es': undefined,
                dashboard: undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall Webpack dependencies',
        task: () => uninstall([...BUILD_DEPENDENCIES, ...WEBPACK_DEPENDENCIES, 'stmux']),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...WEBPACK_DEPENDENCIES),
        optional: ({skipInstall}) => !skipInstall
    },
    {
        text: 'Uninstall Cesium Webpack dependencies',
        task: () => uninstall(CESIUM_WEBPACK_DEPENDENCIES),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...CESIUM_WEBPACK_DEPENDENCIES),
        optional: ({skipInstall, withCesium}) => !skipInstall && withCesium
    }
];
export default addWebpack;