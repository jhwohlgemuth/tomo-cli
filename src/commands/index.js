/* eslint-disable no-magic-numbers */
import {choose, withOptions} from '../utils';
import {
    createPackageJson,
    createSourceDirectory
} from './common';
import addA11y from './add-a11y';
import addBabel from './add-babel';
import {addBrowsersync} from './add-browsersync';
import addEsdoc from './add-esdoc';
import addEslint from './add-eslint';
import addJest from './add-jest';
import addMakefile from './add-makefile';
import addMarionette from './add-marionette';
import {addParcel, removeParcel} from './add-parcel';
import {addPostcss, removePostcss} from './add-postcss';
import {addRollup, removeRollup} from './add-rollup';
import {addWebpack, removeWebpack} from './add-webpack';
import createServer from './create-server';

const createProject = [
    ...createPackageJson,
    ...createSourceDirectory,
    ...addBabel,
    ...addEslint
];
const create = {
    project: createProject,
    app: [
        ...createProject,
        ...addPostcss,
        ...addJest,
        choose({
            default: addMarionette,
            useReact: []// do nothing
        }),
        choose({
            default: addWebpack,
            useRollup: addRollup,
            useParcel: addParcel
        }),
        choose({
            default: addBrowsersync,
            useParcel: []// do nothing
        })
    ],
    server: [
        withOptions({sourceDirectory: '.'}),
        ...createPackageJson,
        ...addEslint,
        ...createServer
    ]
};
const add = {
    a11y: addA11y,
    babel: addBabel,
    browsersync: addBrowsersync,
    esdoc: addEsdoc,
    eslint: [
        ...addBabel,
        ...addEslint
    ],
    jest: [
        ...addBabel,
        ...addJest
    ],
    makefile: addMakefile,
    parcel: [
        ...addBabel,
        ...addParcel
    ],
    postcss: addPostcss,
    rollup: [
        ...addBabel,
        ...addRollup
    ],
    webpack: [
        ...addBabel,
        ...addWebpack
    ]
};
const remove = {
    parcel: removeParcel,
    postcss: removePostcss,
    rollup: removeRollup,
    webpack: removeWebpack
};

export default {
    add,
    remove,
    create,
    new: create// alias for create
};
/**
 * @typedef {Object} task
 * @property {string} text Display text for task
 * @property {function} task Task to execute
 * @property {function} condition Predicate to decide when to execute the task (true) or not (false) - can be async or sync
 * @property {function} [optional] Predicate to decide when to show the task (true) or not (false) - MUST be sync
 */