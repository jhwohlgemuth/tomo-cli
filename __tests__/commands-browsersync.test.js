import {fileContents, run, useTemporaryDirectory} from './tomo-test';
import {createPackageJson} from '../src/commands/common';
import {addPostcss} from '../src/commands/add-postcss';
import {addWebpack} from '../src/commands/add-webpack';
import {addBrowsersync, removeBrowsersync} from '../src/commands/add-browsersync';

jest.mock('is-online', () => (async () => true));

describeOnlyOnLinux('Browsersync', () => {
    let tempDirectory;
    const skipInstall = true;
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    test('Add support', async () => {
        await run(createPackageJson, {});
        await run(addBrowsersync, {skipInstall});
        const noop = fileContents('./package.json');
        await run(addWebpack, {skipInstall});
        await run(addPostcss, {skipInstall});
        await run(addBrowsersync, {skipInstall});
        const packageJson = fileContents('./package.json');
        expect(noop).toMatchSnapshot();
        expect(packageJson).toMatchSnapshot();
    });
    test('Remove support', async () => {
        await run(createPackageJson, {});
        await run(addWebpack, {skipInstall});
        await run(addPostcss, {skipInstall});
        await run(addBrowsersync, {skipInstall});
        await run(removeBrowsersync, {skipInstall});
        const packageJson = fileContents('./package.json');
        expect(packageJson).toMatchSnapshot();
    });
});