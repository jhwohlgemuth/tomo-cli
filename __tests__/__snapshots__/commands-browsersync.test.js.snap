// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browsersync Add support 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`Browsersync Add support 2`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"predeploy\\": \\"npm-run-all clean \\\\\\"build:es -- --mode=production\\\\\\" build:css\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets\\",
        \\"copy:assets\\": \\"cpy \\\\\\"./assets/!(css)/**/*.*\\\\\\" \\\\\\"./assets/**/[.]*\\\\\\" ./output --parents --recursive\\",
        \\"prebuild:es\\": \\"del-cli output/assets\\",
        \\"build:es\\": \\"webpack\\",
        \\"build:stats\\": \\"webpack --mode production --profile --json > stats.json\\",
        \\"build:analyze\\": \\"webpack-bundle-analyzer ./stats.json\\",
        \\"postbuild:es\\": \\"npm run copy\\",
        \\"watch:assets\\": \\"watch \\\\\\"npm run copy\\\\\\" ./assets\\",
        \\"watch:es\\": \\"watch \\\\\\"npm run build:es\\\\\\" ./source\\",
        \\"dashboard\\": \\"webpack-dashboard -- webpack serve --config ./webpack.config.js\\",
        \\"build:css\\": \\"postcss ./assets/css/style.css --dir ./output\\",
        \\"watch:css\\": \\"npm run build:css -- --watch\\",
        \\"prestart\\": \\"npm run build:es\\",
        \\"start\\": \\"npm-run-all --parallel watch:es watch:css serve\\",
        \\"serve\\": \\"browser-sync start --server ./output --files ./output --port 4669\\"
    }
}
"
`;

exports[`Browsersync Remove support 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"predeploy\\": \\"npm-run-all clean \\\\\\"build:es -- --mode=production\\\\\\" build:css\\",
        \\"deploy\\": \\"echo \\\\\\"Not yet implemented - now.sh or surge.sh are supported out of the box\\\\\\" && exit 1\\",
        \\"clean\\": \\"del-cli ./output\\",
        \\"copy\\": \\"npm-run-all --parallel copy:assets\\",
        \\"copy:assets\\": \\"cpy \\\\\\"./assets/!(css)/**/*.*\\\\\\" \\\\\\"./assets/**/[.]*\\\\\\" ./output --parents --recursive\\",
        \\"prebuild:es\\": \\"del-cli output/assets\\",
        \\"build:es\\": \\"webpack\\",
        \\"build:stats\\": \\"webpack --mode production --profile --json > stats.json\\",
        \\"build:analyze\\": \\"webpack-bundle-analyzer ./stats.json\\",
        \\"postbuild:es\\": \\"npm run copy\\",
        \\"watch:assets\\": \\"watch \\\\\\"npm run copy\\\\\\" ./assets\\",
        \\"watch:es\\": \\"watch \\\\\\"npm run build:es\\\\\\" ./source\\",
        \\"dashboard\\": \\"webpack-dashboard -- webpack serve --config ./webpack.config.js\\",
        \\"build:css\\": \\"postcss ./assets/css/style.css --dir ./output\\",
        \\"watch:css\\": \\"npm run build:css -- --watch\\"
    }
}
"
`;
